views/Products/Edit/index.tsx

```diff
@@ -66,6 +61,20 @@ const EditProductForm: FC<MutableProductFormProps> = ({ onSubmit, originalData,
     }
   }
 
+  useEffect(() => {
+    setValue('description', originalData?.description)
+    setValue('name', originalData?.name)
+    setValue('basePriceSol', originalData?.basePriceSol)
+    if (originalData?.imageUrl) setImageUrl(originalData.imageUrl)
```

Comment:
I'm not sure, why not just make this change? Why do you need `getProduct` in `controllers/products.ts`?

Reply:
@Eghinner to keep saving the imageUrl with the previous url so that it does not upload a new file, it used to happen that I clicked on save and the image url was saved as an empty string because it did not upload a file.

Reply:
Yes, you should not upload a new image. And it also caused the empty string problem.
@e-azocar Can you confirm if this is the best way?

Reply:
also, add a check, if the imageURL is valid

Reply:
this may be another issue